### PR TITLE
DX-3099 Update `actions/checkout` and `actions/setup-*`

### PR DESCRIPTION
**Pull request by ajrice6713**
Also Renamed `Validate` to `Test` and added matrix testing

### CODE REVIEW

File: .github/workflows/Test.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+name: Test
```
Comment (ckoegel) [inline .github/workflows/Test.yml:1]: can we name this test with a lowercase t like all of the other test.yml files?

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -9,31 +9,35 @@ jobs:
...
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Set release version
+        run: echo "RELEASE_VERSION=${GITHUB_REF#refs/*/v}" >> $GITHUB_ENV
+
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: "12.x"
```
Comment (ckoegel) [inline .github/workflows/deploy.yml:21]: same as messaging, do we want to try to upgrade these to 16?

File: .github/workflows/Test.yml (hunk 1 above)
Comment (ajrice6713) [inline .github/workflows/Test.yml:1]:
```suggestion
name: test
```

**Review by ckoegel — APPROVED** [review]

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+
+      - name: NPM Install
+        run: npm install
+
+      - name: NPM Build
+        run: npm run build --if-present
```
Comment (bpateldx) [inline .github/workflows/test.yml:30]: 👍 I've not seen this flag used before!

**Review by bpateldx — APPROVED** [review]

**Pull request merged**